datafusion/optimizer/src/single_distinct_to_groupby.rs

```diff
@@ -228,37 +228,18 @@ impl OptimizerRule for SingleDistinctToGroupBy {
                         .collect::<Result<Vec<_>>>()?;
 
                     // construct the inner AggrPlan
```

Comment:
FYI @appletreeisyellow  - this PR contains some changes which may conflict with https://github.com/apache/datafusion/issues/10295